ironfish/src/testReporter/testReporter.ts

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+//import { getConsoleOutput } from '@jest/console'
+import { format } from '@fast-csv/format'
+import { Reporter, TestContext } from '@jest/reporters'
+import { AggregatedResult, Test, TestResult } from '@jest/test-result'
+import { Config } from '@jest/types'
+import { createWriteStream, existsSync, mkdirSync } from 'fs'
+import path from 'path'
+
+type CustomReporter = Pick<Reporter, 'onRunComplete'>
+
+interface Options {
+  outputDirectory: string
+}
+
+export default class TestReporter implements CustomReporter {
+  private globalConfig
+  private reporterConfig: Options = {
+    outputDirectory: '',
+  }
+
+  constructor(globalConfig: Config.GlobalConfig, reporterConfig: Record<string, unknown>) {
+    this.globalConfig = globalConfig
+    const outputDirectory = Object.entries(reporterConfig).find(
+      (entry) => entry[0] === 'outputDirectory',
+    )
+    if (outputDirectory) {
+      this.reporterConfig.outputDirectory = String(outputDirectory[1])
+    }
+  }
+  onRunComplete!: (
+    testContexts: Set<TestContext>,
+    results: AggregatedResult,
+  ) => void | Promise<void>
+
+  public onTestResult(
+    _test: Test,
+    testResult: TestResult,
+    _aggregatedResults: AggregatedResult,
+  ): void {
+    const testFileName = path.parse(testResult.testFilePath).name
+
+    if (!testFileName.includes('perf')) {
+      return
+    }
+
+    if (!existsSync(`${this.globalConfig.rootDir}/${this.reporterConfig.outputDirectory}`)) {
+      mkdirSync(`${this.globalConfig.rootDir}/${this.reporterConfig.outputDirectory}`)
+    }
+
+    const writeStream = createWriteStream(
+      `${this.globalConfig.rootDir}/${this.reporterConfig.outputDirectory}/${testFileName}.csv`,
+    )
+    const stream = format({ headers: true })
+    stream.pipe(writeStream)
+
+    // filter console log
+    const consoleOutputs = testResult.console?.filter((output) => output.type === 'log')
+
+    testResult.testResults.forEach((result, i) => {
+      const row: Record<string, string> = {
+        name: result.title,
+        duration: String(result.duration),
+      }
+
+      if (consoleOutputs && consoleOutputs[i]) {
+        const entries = consoleOutputs[i].message.split(',')
```

Comment:
It doesn't look like there are many better ways to get custom data from the tests into the test reporter. See discussion [here](https://github.com/jestjs/jest/issues/7421). It could be possible but seems like too much extra effort. I think the console buffer is a pretty good potential solution. 

We should definitely make this serialization / deserialization from the console buffer more standardized though. @ygao76 and I talked offline. We're going to merge this in so that we're able to test post-processing steps like uploading to influx but standardizing serialization / deserialization will be done in a future PR.